sovrin_client/agent/endpoint.py

```diff
@@ -19,19 +19,21 @@ def __init__(self, port: int, msgHandler: Callable,
             if ha and ha[1] != port:
                 port = ha[1]
 
-        stackParams = {
+        self.stackParams = {
             "name": name or randomString(8),
             "ha": HA("0.0.0.0", port),
             "main": True,
             "auto": AutoMode.always,
             "mutable": "mutable"
         }
         if basedirpath:
-            stackParams["basedirpath"] = basedirpath
+            self.stackParams["basedirpath"] = basedirpath
 
-        super().__init__(stackParams, self.baseMsgHandler)
+        self._msgHandler = msgHandler
 
-        self.msgHandler = msgHandler
+    def startStack(self):
+        super().__init__(self.stackParams, self.baseMsgHandler)
```

Comment:
Why initialise super here? is done here? Cant this initialisation of super be done in its own initialisation?